geomesa-raster/src/main/scala/org/locationtech/geomesa/raster/data/AccumuloRasterQueryPlanner.scala

```diff
@@ -0,0 +1,158 @@
+    logger.debug(s"Scanning at res: $selectedRes, with hashes: $hashes")
+    val r = hashes.map { gh => modifyHashRange(gh, expectedGeoHashLen, res) }.distinct
+
+    if (r.isEmpty) {
+      logger.debug(s"RasterQueryPlanner: Query was invalid given BBox: ${rq.bbox}")
+      None
+    } else {
+      // of the Ranges enumerated, get the merge of the overlapping Ranges
+      val rows = ARange.mergeOverlapping(r)
+      logger.debug(s"Scanning with ranges: $rows")
+      // setup the RasterFilteringIterator
+      val cfg = new IteratorSetting(90, "raster-filtering-iterator", classOf[RasterFilteringIterator])
+      configureRasterFilter(cfg, AccumuloRasterQueryPlanner.constructRasterFilter(rq.bbox.geom, rasterSft))
+      configureRasterMetadataFeatureType(cfg, rasterSft)
+
+      // TODO: WCS: setup a CFPlanner to match against a list of strings
+      // ticket is GEOMESA-559
+      Some(BatchScanPlan(null, rows, Seq(cfg), Seq.empty[Text], null, -1, hasDuplicates = false))
+    }
+  }
+
+  def selectResolution(suggestedResolution: Double, availableResolutions: List[Double]): Double = {
+    logger.debug(s"RasterQueryPlanner: trying to get resolution $suggestedResolution " +
+      s"from available Resolutions: ${availableResolutions.sorted}")
+    val ret = availableResolutions match {
+      case empty if availableResolutions.isEmpty   => 1.0
+      case one if availableResolutions.length == 1 => availableResolutions.head
+      case _                                       =>
+            val lowerResolutions = availableResolutions.filter(_ <= suggestedResolution)
+            logger.debug(s"RasterQueryPlanner: Picking a resolution from: $lowerResolutions")
+            lowerResolutions match {
+              case Nil => availableResolutions.min
+              case _ => lowerResolutions.max
+            }
+    }
+    logger.debug(s"RasterQueryPlanner: Decided to use resolution: $ret")
+    ret
+  }
+
+  def configureRasterFilter(cfg: IteratorSetting, filter: Filter) = {
+    cfg.addOption(GEOMESA_ITERATORS_ECQL_FILTER, ECQL.toCQL(filter))
+  }
+
+  def configureRasterMetadataFeatureType(cfg: IteratorSetting, featureType: SimpleFeatureType) = {
+    val encodedSimpleFeatureType = SimpleFeatureTypes.encodeType(featureType)
+    cfg.addOption(GEOMESA_ITERATORS_SFT_NAME, rasterSftName)
+    cfg.addOption(GEOMESA_ITERATORS_SIMPLE_FEATURE_TYPE, encodedSimpleFeatureType)
+    cfg.encodeUserData(featureType.getUserData, GEOMESA_ITERATORS_SIMPLE_FEATURE_TYPE)
+  }
+
+}
+
+object AccumuloRasterQueryPlanner {
+  val ff = CommonFactoryFinder.getFilterFactory2
+
+  def constructRasterFilter(geom: Geometry, featureType: SimpleFeatureType): Filter = {
+    val property = ff.property(featureType.getGeometryDescriptor.getLocalName)
+    val bounds = ff.literal(geom)
+    ff.and(ff.intersects(property, bounds), ff.not(ff.touches(property, bounds)))
```

Comment:
There was a specific corner case I noticed when attempting to query for a single tile using the same extent as the tile in which `overlaps` evaluates to true for touching tiles. I think we still have unit tests that look out for that behavior now, but maybe not explicitly.